docs/examples/objc-api-walkthrough/objc-api-walkthrough/ViewController.m

```diff
@@ -0,0 +1,134 @@
+//
+//  ViewController.m
+//  objc-api-walkthrough
+//
+//  Created by James Nocentini on 27/06/2017.
+//  Copyright © 2017 couchbase. All rights reserved.
+//
+
+#import "ViewController.h"
+#include <CouchbaseLite/CouchbaseLite.h>
+
+@interface ViewController ()
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    
+    // create database
+    NSError *error;
+    CBLDatabase* database = [[CBLDatabase alloc] initWithName:@"my-database" error:&error];
+    if (!database) {
+        NSLog(@"Cannot open the database: %@", error);
+    }
+    
+    // create document
+//    NSError* error;
+    CBLDocument* newTask = [[CBLDocument alloc] init];
+    [newTask setObject:@"task-list" forKey:@"type"];
+    [newTask setObject:@"todo" forKey:@"owner"];
+    [newTask setObject:[NSDate date] forKey:@"createAt"];
+    [database saveDocument: newTask error: &error];
+    
+    // mutate document
+    [newTask setObject:@"Apples" forKey:@"name"];
+    [database saveDocument:newTask error:&error];
+    
+    // typed accessors
+    [newTask setObject:[NSDate date] forKey:@"createdAt"];
+    NSDate* date = [newTask dateForKey:@"createdAt"];
+    
+    // database transaction
+    [database inBatch:&error do:^{
+        for (int i = 1; i <= 10; i++)
+        {
+            NSError* error;
+            CBLDocument *doc = [[CBLDocument alloc] init];
+            [doc setObject:@"user" forKey:@"type"];
+            [doc setObject:[NSString stringWithFormat:@"user %d", i] forKey:@"name"];
+            [database saveDocument:doc error:&error];
+            NSLog(@"saved user document %@", [doc stringForKey:@"name"]);
+        }
+    }];
+    
+    // blob
+    UIImage *image = [UIImage imageNamed:@"avatar.jpg"];
+    NSData *data = UIImageJPEGRepresentation(image, 1);
+    
+    CBLBlob *blob = [[CBLBlob alloc] initWithContentType:@"image/jpg" data:data];
+    [newTask setObject:blob forKey: @"avatar"];
+    
+//    NSError* error;
+    [database saveDocument: newTask error:&error];
+    if (error) {
+        NSLog(@"Cannot save document %@", error);
+    }
+    
+    CBLBlob* taskBlob = [newTask blobForKey:@"avatar"];
+    UIImage* taskImage = [UIImage imageWithData:taskBlob.content];
+    
+    // query
+    CBLQuery* query = [CBLQuery select:[CBLQuerySelect all]
```

Comment:
We have changed this API in DB12 as we support projection, and We don't support SELECT * yet as of DB12. The same behavior here will be the following:

```
[CBLQuery select:@[]
```
This means that the statement projects nothing.